### ISSUE TITLE
/mnt/ncdata not mounted after reboot

### ISSUE THREAD

**Issue by nasi87** [nextcloud/vm#769]
## Steps to reproduce
1. install clean Ubuntu
2. sudo bash -c "$(wget -q -O - https://raw.githubusercontent.com/nextcloud/vm/master/nextcloud_install_production.sh)" 
3. reboot (automatic at end of script)

## Expected behaviour
after reboot proceed with nextcloud-startup-script.sh

## Actual behaviour
nextcloud-startup-script.sh aborts because of missing /mnt/ncdata

`ncadmin@nextcloud:~$ zpool list`
`no pools available`

temporary workaround:
`root@nextcloud:~# zpool import ncdata`

but it doesn't survive next reboot

## Server configuration

**Nextcloud server version:** none

**Server OS** Ubuntu 18.04.1 server amd64

**How did you install the VM?** (Scripted install from master **OR** Released version)
script from master

## Network
**Do you use DHCP?**
yes
**Is port 80 and/or 443 open?**
not yet, but afaik not necessary
## Logs / Screenshots

will provide if needed

#### Log file (/mnt/ncdata/nextcloud.log)
```
Insert your nextcloud.log file here
```

#### Installation log (command output)
```
Insert the CLI output here
```

**Comment by nasi87** [comment]
PS: /dev/sdb - which shall be DATA drive - is an iSCSI drive on my NAS

**Comment by enoch85** [comment]
Thanks for your report.

> PS: /dev/sdb - which shall be DATA drive - is an iSCSI drive on my NAS

Well, there I think you have the issue - your iSCSI setup.

Can you please confirm that it's the same behaviour if you not mount iSCSI and just run it plain as intended?

The scripts are not designed to cover all scenarios, but it should be possible to mount iSCSI if done right, never tested though.

On my servers I run a NFS mount and I never had any issues with that.

**Comment by enoch85** [comment]
Just tested here on the latest version and there are no issues across reboots.

What the script does is basically:
1. [Unmount if mounted](https://github.com/nextcloud/vm/blob/master/static/format-sdb.sh#L22) (/dev/sdb)
2. [Formats and mounts](https://github.com/nextcloud/vm/blob/master/static/format-sdb.sh#L124-L140) it as ZFS as with any other drive, but in this case it's ZFS specific.

The rest is just a bunch of checks and preconditions.

So I belive this "bug" is one of two things.
1. A faulty iSCSI setup
2. A bug with the ZFS packages in Ubuntu.

Closing since it works as expected. Please reopen if you think I'm wrong.

**Comment by enoch85** [comment]
Hi @nasi87 

So, it seems you are right after all. ZFS can be a bit buggy, as I noticed in Ubuntu 20.04. Sorry for not really believing you the last time. 

In an effort to trying to fix it, I've been working on this the past day: https://github.com/nextcloud/vm/blob/master/static/change-to-zfs-mount-generator.sh

Do you still have issues? If yes, how did you solve it? Also, please try the script, please remember to make a snapshot first.

Thanks!

**Comment by enoch85** [comment]
Related: https://github.com/nextcloud/vm/issues/1204

**Comment by nasi87** [comment]
> Hi @nasi87
> 
> So, it seems you are right after all. ZFS can be a bit buggy, as I noticed in Ubuntu 20.04. Sorry for not really believing you the last time.
> 

Thanks for your feedback. 
And no worry, I thought it was iSCSI-related too. So I did change the way how to mount this iSCSI-LUN. Instead of mounting it in Nextcloud-VM directly it is now mounted by my Proxmox-host which then mounts it as a virtual disk. So Nextcloud-VM sees a normal virtual disk as usual. 

So unfortunately I did not change anything ZFS-related.
